File: call-management/call-blocking/update-blocked-allowed-number/code-samples/updateBlockedAllowedNumber.php
<?php
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// PATH PARAMETERS
$accountId = '<ENTER VALUE>';
$extensionId = '<ENTER VALUE>';
$blockedNumberId = '<ENTER VALUE>';

// POST BODY
$body = array(
    'phoneNumber' => '<ENTER VALUE>',
    'label' => '<ENTER VALUE>',
    'status' => 'Blocked'
)

require('vendor/autoload.php');
$rcsdk = new RingCentral\SDK\SDK(getenv('clientId'), getenv('clientSecret'), getenv('serverURL'));
$platform = $rcsdk->platform();
$platform->login(getenv('username'), getenv('extension'), getenv('password'));
$r = $platform->put("/restapi/v1.0/account/{$accountId}/extension/{$extensionId}/caller-blocking/phone-numbers/{$blockedNumberId}", $body);
?>

File: call-management/call-blocking/update-blocked-allowed-number/code-samples/updateBlockedAllowedNumber.cs
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// PATH PARAMETERS
string accountId = "<ENTER VALUE>";
string extensionId = "<ENTER VALUE>";
string blockedNumberId = "<ENTER VALUE>";

// POST BODY
AddBlockedAllowedPhoneNumber addBlockedAllowedPhoneNumber = new AddBlockedAllowedPhoneNumber {
    phoneNumber = "<ENTER VALUE>",
    label = "<ENTER VALUE>",
    status = 'Blocked'
};

RestClient rc = new RestClient(
    Environment.GetEnvironmentVariable("clientId"),
    Environment.GetEnvironmentVariable("clientSecret"),
    false
);
await rc.Authorize(
    Environment.GetEnvironmentVariable("username"),
    Environment.GetEnvironmentVariable("extension"),
    Environment.GetEnvironmentVariable("password")
);
var r = await rc.Restapi().Account(accountId).Extension(extensionId).CallerBlocking().PhoneNumbers(blockedNumberId).Put(addBlockedAllowedPhoneNumber);
// PROCESS RESPONSE
